models/curvefi/ethereum/curvefi_ethereum_trades.sql

```diff
@@ -0,0 +1,745 @@
+{% set usdc_ethereum_token = "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48" %}
+{% set usdt_ethereum_token = "0xdac17f958d2ee523a2206206994597c13d831ec7" %}
+{% set proxy_susd_token = "0x57ab1ec28d129707052df4df418d58a2d46d5f51" %}
+
+{% set tusd_token = "0x0000000000085d4780b73119b644ae5ecd22b376" %}
+{% set busd_token = "0x4fabb145d64652a948d72533023f6e7a623c7c53" %}
+
+{% set ysusd_token = "0xf61718057901f84c4eec4339ef8f0d86d2b45600" %}
+{% set ycrv_token = "0xdf5e0e81dff6faf3a7e52ba697820c5e32d806a8" %}
+
+{% set cdai_token = "0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643" %}
+{% set cusdc_token = "0x39aa39c021dfbae8fac545936693ac917d5e7563" %}
+
+{% set ydai_token = "0x16de59092dae5ccf4a1e6439d611fd0653f0bd01" %}
+{% set yusdc_token = "0xd6ad7a6750a7593e092a9b218d66c0a814a3436e" %}
+{% set yusdt_token = "0x83f798e925bcd4017eb265844fddabb448f1707d" %}
+{% set ytusd_token = "0x73a052500105205d34daf004eab301916da8190f" %}
+
+{% set ydai_token3 = "0xc2cb1040220768554cf699b0d863a3cd4324ce32" %}
+{% set yusdc_token3 = "0x26ea744e5b887e5205727f55dfbe8685e3b21951" %}
+{% set yusdt_token3 = "0xe6354ed5bc4b393a5aad09f21c46e101e692d447" %}
+{% set ybusd_token = "0x04bc0ab673d88ae9dbc9da2380cb6b79c4bca9ae" %}
+
+{% set ycdai_token = "0x99d1fa417f94dcd62bfe781a1213c092a47041bc" %}
+{% set ycusdc_token3 = "0x9777d7e2b60bb01759d0e2f8be2095df444cb07e" %}
+{% set ycusdt_token3 = "0x1be5d71f2da660bfdee8012ddc58d024448a0a59" %}
+{% set usdp_token = "0x8e870d67f660d95d5be530380d0ec0bd388289e1" %}
+
+{% set renBTC_ethereum_token = "0xeb4c2781e4eba804ce9a9803c67d0893436bb27d" %}
+{% set wbtc_ethereum_token = "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599" %}
+{% set sBTC_ethereum_token = "0xfe18be6b3bd88a2d2a7f928d00292e7a9963cfc6" %}
+
+{% set weth_contract = "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2" %}
+{% set steth_contract = "0xae7ab96520de3a18e5e111b5eaab095312d7fe84" %}
+
+WITH dexs AS
+(
+    -- Curvefi TokenExchange
+    SELECT
+        evt_block_time AS block_time,
+        '' AS version,
+        buyer AS taker,
+        '' AS maker,
+        tokens_bought AS token_bought_amount_raw,
+        tokens_sold AS token_sold_amount_raw,
+        NULL AS amount_usd,
+        CASE
+            WHEN bought_id = 0 THEN '{{dai_ethereum_token}}'
+            WHEN bought_id = 1 THEN '{{usdc_ethereum_token}}'
+            WHEN bought_id = 2 THEN '{{usdt_ethereum_token}}'
+            WHEN bought_id = 3 THEN '{{proxy_susd_token}}'
+        END as token_bought_address,
+        CASE
+            WHEN sold_id = 0 THEN '{{dai_ethereum_token}}'
+            WHEN sold_id = 1 THEN '{{usdc_ethereum_token}}'
+            WHEN sold_id = 2 THEN '{{usdt_ethereum_token}}'
+            WHEN sold_id = 3 THEN '{{proxy_susd_token}}'
+        END as token_sold_address,
+        contract_address AS project_contract_address,
+        evt_tx_hash AS tx_hash,
+        '' AS trace_address,
+        evt_index
+    FROM {{ source('curvefi_ethereum', 'susd_swap_evt_TokenExchange') }}
```

Comment:
Could you add incremental filters on decoded tables ?

Here, it would be:
    `{% if is_incremental() %}
    AND evt_block_time >= date_trunc("day", now() - interval '1 week')
    {% endif %}`
    
    But this comment applies to all the other sources

Reply:
[Added](https://github.com/duneanalytics/spellbook/pull/1583/commits/2f5a60a83289298e2384d0d4fedac5a6fab59fc9)